code/modules/atmospherics/machinery/portable/canister.dm

```diff
@@ -287,31 +287,29 @@
 	air_contents.gases[/datum/gas/nitrogen][MOLES] = (N2STANDARD * maximum_pressure * filled) * air_contents.volume / (R_IDEAL_GAS_EQUATION * air_contents.temperature)
 
 /obj/machinery/portable_atmospherics/canister/update_icon_state()
-	if(machine_stat & BROKEN)
+	if(machine_stat & BROKEN && !(splittext(icon_state,  "-").len > 1))
```

Comment:
What's this nonsense?

Reply:
it's testing if this proc has been run before, to prevent stacking -'s. I can't remember the exact proc call chain, but it's some icon related proc calling update_icon_state more then it should, and I couldn't figure out any good way to clean it up.

Reply:
Just add a baseline icon state variable.

Reply:
Thanks a lot